README.md

```diff
@@ -1 +1,90 @@
+```bash
+docker run --network=host -v <absolute path to config>/prometheus.yml:/etc/prometheus/prometheus.yml prom/prometheus
+```
+
+## Starting Prometheus via Local Clone
+Alternatively you can checkout Prometheus source code and run it locally
+```bash
+git clone https://github.com/prometheus/prometheus.git
+cd prometheus
+```
+Configure prometheus.yml to scrape from the fhir_target.
+For our current purposes the example prometheus configuration at `prometheus/documentation/examples/prometheus.yml` will suffice
+#### Adding the FHIR Querier service as a target
+Make sure the config file contains the following:
+```yaml
+scrape_configs:
+  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
+
+  - job_name: 'FHIRQUERY'
+
+    # metrics_path defaults to '/metrics'
+    # scheme defaults to 'http'.
+
+    static_configs:
+    - targets: ['localhost:8080']
+```
+Finally, you can start an instance of prometheus by running:
+```bash
+./prometheus --config.file=prometheus.yml
+```
+
+## Starting Grafana
+Make sure that you have Docker installed and running on your machine
+```bash
+docker run -d -p 3000:3000 grafana/grafana
+```
+
+## Viewing Colllected Data In Grafana
+1. Navigate to `http://localhost:3000/` in a web browser
+2. Login using Username: admin, Password admin
+3. Add a Prometheus datasource, running on `http://localhost:9090` by default. Select access Browser and then Save
+4. From the main page create a Dashboard, adding visualizations for the metrics you would like to explore
+
+# Testing
+### Running All Unit Tests
+To run all tests in a project run:
+```bash
+go test ./...
+```
+This will search the current directory and all sub directories for files matching the pattern `*_test.go`. You also have the option to specify a package location or file if you do not want to run all tests at once.
+### Running Tests With Coverage
+If you are interested in having coverage information displayed along with the pass/fail status' of the tests you can run the tests with the `--covermode=count` option.
+```bash
+go test -covermode=count ./...
+ok      command-line-arguments  0.009s  coverage: 31.8% of statements
+```
+If you are interested in more in-depth coverage analysis you'll have to generate a coverage report, in the following command the coverage report is named coverage.out
+```bash
+go test -coverprofile=coverage.out ./...
+```
+Using the generated coverage file you can use `go tool cover` to view the coverage report in your browser and see which lines are or are not being covered.
+```bash
+go tool cover -html=coverage.out
+```
+
+# Contributing
+## Lintr
+Code going through PR should pass the lintr invoked by running:
+```bash
+golangci-lint run
```

Comment:
thoughts on changing this to `golangci-lint run -E gofmt`?

This enables a check to confirm that `gofmt -s` has been run. (info here: https://golang.org/cmd/gofmt/)
So before you'd run this, you'd also run `gofmt -s -w`

See the info on the linters here: https://github.com/golangci/golangci-lint#quick-start